_plugins/docs_generator.rb

```diff
@@ -141,6 +142,28 @@ def convert_with_sphinx(repo_name, repo_path, repo_data)
       end
       repo_content[permalink] = json_content
     end
-    repo_content.sort {|a, b| a[0].length <=> b[0].length}
+    repo_content.sort do |a, b|
+      first_depth = a[0].count('/')
+      second_depth = b[0].count('/')
+      if first_depth == second_depth
+        first_sourcename = a[1]['sourcename'] || ''
+        first_order = repo_index_pattern.index do |pattern|
+          File.fnmatch?(pattern, first_sourcename)
+        end || -1
+        second_sourcename = b[1]['sourcename'] || ''
+        second_order = repo_index_pattern.index do |pattern|
+          File.fnmatch?(pattern, second_sourcename)
+        end || -1
+        if first_order == second_order
+          first_title = a[1]['title'] || ''
+          second_title = b[1]['title'] || ''
+          first_title <=> second_title
+        else
+          first_order <=> second_order
+        end
+      else
+        first_depth <=> second_depth
```

Comment:
I am wondering why the depth of a navigation item should have any factor in determining the order? That sounds like a pretty arbitrary assumption. E.g if there are sibling items "Foo" and "Bar" I would expect them to be ordered alphabetically and not differently just because "Bar" has a deeper nesting. If in the future additional subpages are created which increase the depth that would also imply that all of a sudden the position of the item in the navigation changes.

Reply:
>if there are sibling items "Foo" and "Bar" I would expect them to be ordered alphabetically and not differently just because "Bar" has a deeper nesting

That's not what that `depth` means in this context. It is the depth of the page w.r.t. the project source root, not the depth of the subtree rooted at said page. For instance, `foo` and `bar` will have the same depth if they are sibling pages. 

As to why that ordering, it is to make it easier for calling code to infer a hierarchy, but it has no impact on how the index is rendered.